src/sentry/api/endpoints/project_artifact_bundle_file_details.py

```diff
@@ -0,0 +1,103 @@
+        except Exception:
+            # In case we fail here, we want to close the fail before returning.
+            archive.close()
+            return Response(
+                {"error": f"The file {file_path} can't be found in the artifact bundle"},
+                status=404,
+            )
+
+        response = FileResponse(
+            ClosesDependentFiles(fp, archive),
+            content_type=headers.get("content-type", "application/octet-stream"),
+        )
+        response["Content-Length"] = file_info.file_size if file_info is not None else None
+        response["Content-Disposition"] = 'attachment; filename="%s"' % posixpath.basename(
+            " ".join(file_path.split())
+        )
+
+        return response
+
+
+@region_silo_endpoint
+class ProjectArtifactBundleFileDetailsEndpoint(
+    ProjectEndpoint, ProjectArtifactBundleFileDetailsMixin
+):
+    permission_classes = (ProjectReleasePermission,)
+
+    def get(self, request: Request, project, bundle_id, file_id) -> Response:
+        """
+        Retrieve the file of an artifact bundle
+        `````````````````````````````````
+
+        Return details on an individual file within a release.  This does
+        not actually return the contents of the file, just the associated
+        metadata.
+
+        :pparam string organization_slug: the slug of the organization the
+                                          release belongs to.
+        :pparam string project_slug: the slug of the project to retrieve the
+                                     file of.
+        :pparam string bundle_id: the bundle_id of the artifact bundle that
+                                    should contain the file identified by file_id.
+        :pparam string file_id: the ID of the file to retrieve.
+        :auth: required
+        """
+        if not has_download_permission(request, project):
+            return Response(
+                {"error": "You don't have the permissions to download this file"}, status=403
+            )
+
+        try:
+            file_path = base64.urlsafe_b64decode(file_id).decode()
+        except Exception:
+            return Response(
+                {"error": f"The file_id {file_id} is invalid"},
+                status=400,
+            )
+
+        try:
+            project_artifact_bundle = ProjectArtifactBundle.objects.filter(
+                project_id=project.id, artifact_bundle__bundle_id=bundle_id
+            ).select_related("artifact_bundle__file")[0]
+        except IndexError:
+            return Response(
+                {
+                    "error": f"The artifact bundle with {bundle_id} is not bound to this project or doesn't exist"
+                },
+                status=400,
+            )
+
+        artifact_bundle = project_artifact_bundle.artifact_bundle
+
+        try:
+            archive = ArtifactBundleArchive(artifact_bundle.file.getfile(), build_memory_map=False)
+        except Exception:
```

Comment:
do you want to silently catch everything? or may be try to add sentry.capture_exception or specify one?

Reply:
I could capture it on sentry, good idea!